src/Drivers/DPDK/DpdkDriverImpl.cc

```diff
@@ -179,7 +177,6 @@ Driver::Packet*
 DpdkDriver::Impl::allocPacket()
 {
     DpdkDriver::Impl::Packet* packet = nullptr;
-    SpinLock::Lock lock(packetLock);
     static const int MBUF_ALLOC_LIMIT = NB_MBUF - NB_MBUF_RESERVED;
     if (mbufsOutstanding < MBUF_ALLOC_LIMIT) {
```

Comment:
I decided to make the thread-safety of `ObjectPool` configurable via a template argument, which is generally useful anyway. Since I am lazy, I am going to revert all changes on DpdkDriver.